tree/30_generic_methods/logger_rudder.cf

```diff
@@ -29,9 +29,11 @@ bundle agent logger_rudder(message, class_prefix)
 {
   vars:
 
+
       "expected_reports_source" string => "${sys.workdir}/inputs/rudder_expected_reports.csv";
       "expected_reports_temp"   string => "${expected_reports_source}.tmp";
-      "expected_reports_file"   string => "${expected_reports_source}.res";
+      # We add the PID to the file name to be able to handle cuncurrent agent runs with separate expected reports
+      "expected_reports_file"   string => "${sys.workdir}/state/rudder_expected_reports.${this.promiser_pid}.csv.res";
```

Comment:
Could you also add a comment here as to why we're placing the file in a different directory? (to avoid purging during update)